.github/workflows/release.yaml

```diff
@@ -0,0 +1,338 @@
+        with:
+          submodules: recursive
+      - run: |
+          export VERSION="${{ needs.version.outputs.VERSION }}"
+          BUILD_ARCH=amd64 ./scripts/build_linux.sh
+          BUILD_PLATFORM=linux/amd64 IMAGE_NAME=ollama/release ./scripts/build_docker.sh
+          # TODO push images
+          mv dist/deps/* dist/
+      - uses: actions/upload-artifact@v4
+        with:
+          name: dist-linux-amd64
+          path: |
+            dist/*linux*
+            !dist/*-cov
+
+  # Linux ARM assets built using the container based build
+  # (at present, docker isn't pre-installed on arm ubunutu images)
+  build-linux-arm64:
+    needs:
+      - version
+    runs-on: linux-arm64
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: recursive  
+      - name: "Install Docker"
+        run: |
+          # Add Docker's official GPG key:
+          env
+          uname -a
+          sudo apt-get update
+          sudo apt-get install -y ca-certificates curl
+          sudo install -m 0755 -d /etc/apt/keyrings
+          sudo curl -fsSL https://download.docker.com/linux/ubuntu/gpg -o /etc/apt/keyrings/docker.asc
+          sudo chmod a+r /etc/apt/keyrings/docker.asc
+
+          # Add the repository to Apt sources:
+          echo \
+            "deb [arch=$(dpkg --print-architecture) signed-by=/etc/apt/keyrings/docker.asc] https://download.docker.com/linux/ubuntu \
+            $(. /etc/os-release && echo "$VERSION_CODENAME") stable" | \
+            sudo tee /etc/apt/sources.list.d/docker.list > /dev/null
+          sudo apt-get update
+          sudo apt-get install -y docker-ce docker-ce-cli containerd.io
+          sudo usermod -aG docker $USER
+          sudo apt-get install acl
+          sudo setfacl --modify user:$USER:rw /var/run/docker.sock
+      - run: |
+          export VERSION="${{ needs.version.outputs.VERSION }}"
+          BUILD_ARCH=arm64 sh -x ./scripts/build_linux.sh
+          BUILD_PLATFORM=linux/arm64 IMAGE_NAME=ollama/release ./scripts/build_docker.sh
+          # TODO push images
+      - uses: actions/upload-artifact@v4
+        with:
+          name: dist-linux-arm64
+          path: |
+            dist/*linux*
+            !dist/*-cov
+
+  # Aggregate all the assets and ship a release
+  release: 
+    needs:
+      - version
+      - build-darwin
+      - build-windows
+      - build-linux-amd64
+      - build-linux-arm64
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Retrieve built artifact
+        uses: actions/download-artifact@v4
+        with:
+          path: dist
+          pattern: dist-*
+          merge-multiple: true
+      # TODO verify the payloads look right here
+      - run: "ls -lh dist/"
+
+      # TODO - combine the docker image assets into a multiarch manifest (non-rocm)
+
+      # TODO uncomment this when we're ready to actually generate and ship a release
+      # - uses: ncipollo/release-action@v1

```

Comment:
You don't need this, just use `gh` which is available on the runner

Reply:
Can you elaborate?

Reply:
gh is github's cli. it comes preinstalled on all default runners. you can use it to manipulate the release: https://cli.github.com/manual/gh_release. you'll need to set `GH_TOKEN`

Reply:
From the docs, it looks like I'd have to implement some of the idempotent logic I'm planning to leverage from this action.  My goal was to be able to pre-create the draft release with release notes, or push the tag in either order and have it work without throwing away anything written by a human.  I'm not opposed to reinventing that logic, but my preference is to stick with the action which handles this already.